staging/src/k8s.io/apiextensions-apiserver/test/integration/versioning_test.go

```diff
@@ -136,3 +140,266 @@ func testStoragedVersionInCRDStatus(t *testing.T, ns string, noxuDefinition *api
+	noxuDefinition := testserver.NewNoxuCustomResourceDefinition(apiextensionsv1beta1.NamespaceScoped)
+	noxuDefinition, err = testserver.CreateNewCustomResourceDefinition(noxuDefinition, apiExtensionClient, dynamicClient)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// ensure v1beta1 appears in discovery
+	group := "mygroup.example.com"
+	if err := wait.Poll(100*time.Millisecond, wait.ForeverTestTimeout, func() (done bool, err error) {
+		_, err = apiExtensionClient.Discovery().ServerResourcesForGroupVersion(group + "/v1beta1")
+		if err == nil {
+			return true, nil
+		}
+		if errors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}); err != nil {
+		t.Fatalf("failed to see v1beta1 in discovery: %v", err)
+	}
+
+	// create a CR and make sure it's version is v1beta1
+	ns := "not-the-default"
+	noxuV1beta1ResourceClient := newNamespacedCustomResourceClient(ns, dynamicClient, noxuDefinition)
+	noxuInstanceToCreate := testserver.NewNoxuInstance(ns, "foo")
+	createdFooInstance, err := noxuV1beta1ResourceClient.Create(noxuInstanceToCreate)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if createdFooInstance.GetAPIVersion() != "mygroup.example.com/v1beta1" {
+		t.Fatalf("invalid version: expected \"mygroup.example.com/v1beta1\", got %v", createdFooInstance.GetAPIVersion())
+	}
+
+	// add v1beta2 with served = true and storage = false
+	gottenCRD, err := apiExtensionClient.ApiextensionsV1beta1().CustomResourceDefinitions().Get(noxuDefinition.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("could not get CRD: %v", err)
+	}
+	v1beta2 := apiextensionsv1beta1.CustomResourceDefinitionVersion{
+		Name:    "v1beta2",
+		Served:  true,
+		Storage: false,
+	}
+	gottenCRD.Spec.Versions = append(gottenCRD.Spec.Versions, v1beta2)
+	err = wait.Poll(500*time.Millisecond, wait.ForeverTestTimeout, func() (bool, error) {
+		updatedCRD, err := apiExtensionClient.ApiextensionsV1beta1().CustomResourceDefinitions().Update(gottenCRD)
+		if err != nil {
+			return false, err
+		}
+
+		if len(updatedCRD.Spec.Versions) == 2 {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		t.Fatalf("could not update CRD: %v", err)
+	}
+
+	// ensure both v1beta1 and v1beta2 appear in discovery
+	if err := wait.Poll(100*time.Millisecond, wait.ForeverTestTimeout, func() (done bool, err error) {
+		_, err1 := apiExtensionClient.Discovery().ServerResourcesForGroupVersion(group + "/v1beta1")
+		_, err2 := apiExtensionClient.Discovery().ServerResourcesForGroupVersion(group + "/v1beta2")
+		if err1 == nil && err2 == nil {
+			return true, nil
+		}
+		if errors.IsNotFound(err1) || errors.IsNotFound(err2) {
+			return false, nil
+		}
+		return false, fmt.Errorf("failed to see both v1beta1 and v1beta2 in discovery")
+	}); err != nil {
+		t.Fatalf("failed to see both v1beta1 and v1beta2 in discovery: %v", err)
+	}
+
+	// ensure that both v1beta1 and v1beta2 are usable by
+	// ensuring that the old CR still has version = v1beta1
```

Comment:
when you get the old one from old client, it will be v1beta1 if you get it from new client the same old CR will have v1beta2. That's what our conversion do.